### PR TITLE
Add tree-sitter-preproc crate

### PR DESCRIPTION
**Pull request by Luni-4**
This PR adds `tree-sitter-preproc` grammar as a `Rust` crate to `rust-code-analysis`

Thanks in advance for your review! :)

### CODE REVIEW

File: tree-sitter-preproc/.gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+Cargo.lock
```
Comment (marco-c) [inline tree-sitter-preproc/.gitignore:1]: Why ignoring Cargo.lock here?

File: tree-sitter-preproc/bindings/rust/lib.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// -*- coding: utf-8 -*-
```
Comment (marco-c) [inline tree-sitter-preproc/bindings/rust/lib.rs:1]: Same question here, did you copy this from somewhere, or did you write it from scratch?

File: tree-sitter-preproc/bindings/rust/build.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+use std::path::Path;
```
[the remark this answers is not in the document]
Comment (Luni-4) [reply tree-sitter-preproc/bindings/rust/build.rs:1]: I used the same style adopted upstream, in order to be coherent with the other grammar releases.

File: tree-sitter-preproc/.gitignore (hunk 1 above)
Comment (Luni-4) [reply tree-sitter-preproc/.gitignore:1]: Considering the hypothesis to move that grammar in an independent repository

File: tree-sitter-preproc/bindings/rust/build.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (marco-c) [reply tree-sitter-preproc/bindings/rust/build.rs:1]: Please add a reference to the original, `// Adapted from XXX`

File: tree-sitter-preproc/bindings/rust/lib.rs (hunk 2 above)
Comment (marco-c) [reply tree-sitter-preproc/bindings/rust/lib.rs:1]: Same here

File: tree-sitter-preproc/bindings/rust/build.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Luni-4) [reply tree-sitter-preproc/bindings/rust/build.rs:2]: Done

File: tree-sitter-preproc/bindings/rust/lib.rs (hunk 2 above)
Comment (Luni-4) [reply tree-sitter-preproc/bindings/rust/lib.rs:2]: Done

**Review by marco-c — APPROVED** [review]

**Pull request merged**